### PR TITLE
[CAN-3237] Fixes how body & header keys look like on json logs

### PR DESCRIPTION
**Pull request by klesgidis**
## Summary
Fixes how body & header keys look like on json logs

### SimlpleFormatter
Before
```
[2020-09-23T16:48:53.676Z] [c7db6700-fd1b-4982-8619-08bc0a7a2ded] --> POST /test 200 8ms 20b | request.headers.x-request-id="c7db6700-fd1b-4982-8619-08bc0a7a2ded", request.body.candidate.firstname="kyriakos", request.body.candidate.lastname="lesgidis"
```
After
```
[2020-09-23T17:00:41.429Z] [58f56ada-63ec-462e-ab39-169fa33f24e4] --> POST /test 200 5ms 20b | request.headers.x-request-id="58f56ada-63ec-462e-ab39-169fa33f24e4", request.body.candidate.firstname="kyriakos", request.body.candidate.lastname="lesgidis"
```

### ExtendedFormatter
Before
```
[2020-09-23T16:49:59.084Z] [b30b86f3-6b08-47da-ae68-d9a0df142dd7] --> POST /test 200 6ms 20b | status=200, duration=6, requestId="b30b86f3-6b08-47da-ae68-d9a0df142dd7", method="POST", protocol="http", path="/test", request.headers.x-request-id="b30b86f3-6b08-47da-ae68-d9a0df142dd7", request.body.candidate.firstname="kyriakos", request.body.candidate.lastname="lesgidis", log_tag="outbound_response", contentLength=20, userAgent="PostmanRuntime/7.26.5"
```
After
```
[2020-09-23T16:57:08.250Z] [0a4b907d-4d1f-4941-9aa0-4ddc28005520] --> POST /test 200 12ms 20b | status=200, duration=12, requestId="0a4b907d-4d1f-4941-9aa0-4ddc28005520", method="POST", protocol="http", path="/test", request.headers.x-request-id="0a4b907d-4d1f-4941-9aa0-4ddc28005520", request.body.candidate.firstname="kyriakos", request.body.candidate.lastname="lesgidis", log_tag="outbound_response", contentLength=20, userAgent="PostmanRuntime/7.26.5"
```

### JsonFormatter
Before
```
{"severity":"info","timestamp":"2020-09-23T17:03:50.180Z","httpRequest":{"requestMethod":"POST","requestUrl":"/test","status":200,"protocol":"http","latency":"0.006000s","responseSize":20,"userAgent":"PostmanRuntime/7.26.5"},"logging.googleapis.com/operation":{"id":"ed5fa096-74f8-46fa-ace5-fc8a519c3b46"},"request.body.candidate.firstname":"kyriakos","request.body.candidate.lastname":"lesgidis","request":{"headers":{"x-request-id":"ed5fa096-74f8-46fa-ace5-fc8a519c3b46"}},"headers":{},"requestId":"ed5fa096-74f8-46fa-ace5-fc8a519c3b46","query":null,"log_tag":"outbound_response"}
```
After
```
{"severity":"info","timestamp":"2020-09-23T20:33:00.217Z","httpRequest":{"requestMethod":"POST","requestUrl":"/test","status":200,"protocol":"http","latency":"0.007000s","responseSize":20,"userAgent":"PostmanRuntime/7.26.5"},"logging.googleapis.com/operation":{"id":"59c08aab-0e10-4b21-bdad-17fa10dd4b3c"},"request":{"body":{"candidate":{"firstname":"kyriakos","lastname":"lesgidis"}},"headers":{"x-request-id":"59c08aab-0e10-4b21-bdad-17fa10dd4b3c"}},"headers":{},"requestId":"59c08aab-0e10-4b21-bdad-17fa10dd4b3c","query":null,"log_tag":"outbound_response"}
```

The above requests are after testing `riviere` manually

### CODE REVIEW

File: lib/formatters/JsonFormatter.js [hunk 1]
```diff
@@ -60,11 +61,10 @@ class JsonFormatter extends BaseFormatter {
         id: obj.requestId
       }
     };
-
+    const meta = merge(metaBody, metaHeaders);
```
Comment (nikostoulas) [inline lib/formatters/JsonFormatter.js:64]: ❓ Why are you merging body with headers?

File: lib/formatters/JsonFormatter.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+const { merge } = require('lodash');
```
Comment (SokratisVidros) [inline lib/formatters/JsonFormatter.js:1]: 🙃 ```suggestion
const merge = require('lodash/merge');
```

File: lib/utils.js [hunk 3]
```diff
@@ -22,7 +20,11 @@ function extractHeaders(headersRegex, headers, prefix = '') {
   });
 
   if (Object.keys(metaHeaders).length) {
-    metaHeaders = flat({ [`${prefix + (prefix && '.')}headers`]: metaHeaders });
+    if (prefix) {
+      metaHeaders = { [prefix]: { headers: metaHeaders } };
```
Comment (nikostoulas) [inline lib/utils.js:24]: 👍

**Review by nikostoulas — APPROVED** [review]
👏  Nice work @klesgidis

**Review by SokratisVidros — APPROVED** [review]
@klesgidis What's the difference between before and after? Text diffing shows they are identical. Am I missing something?

File: lib/formatters/JsonFormatter.js (hunk 1 above)
Comment (klesgidis) [reply lib/formatters/JsonFormatter.js:64]: I spent 30' yesterday to find ouy what was going on here. 
metaBody has value `{request: {body: {...}}}` and metaHeaders has value `{request: {headers: {...}}}`, so after extended the two objects, headers were overriding the body object.

So I had to merge

File: lib/formatters/JsonFormatter.js (hunk 2 above)
Comment (klesgidis) [reply lib/formatters/JsonFormatter.js:1]: The first 2 formatters MUST be identical to be backwards compatible. 

The only difference is on JSON Formatter. We can discuss if you want offline the benefits of having a json object in json logs and why I need it like this

**Review by klouvas — APPROVED** [review]

**Pull request merged**